packages/lodestar/src/sync/gossip/handler.ts

```diff
@@ -74,7 +74,7 @@ export class BeaconGossipHandler implements IGossipHandler {
   };
 
   private onAggregatedAttestation = async (aggregate: SignedAggregateAndProof): Promise<void> => {
-    await this.chain.receiveAttestation(aggregate.message.aggregate);
+    await this.db.aggregateAndProof.add(aggregate.message);
```

Comment:
need confirmation

Reply:
I think this should still be passed to the chain?
We need to process incoming attestations for the fork choice

Reply:
I'm not sure, currently we process attestations from received block so I thought this is to put it to the OpPool only to produce block and once the block is published we'll finally process attestations from block.

Reply:
if we pass to the chain here we may process again when the block containing the same attestation is published, isn't it a duplicate?

Reply:
the fork choice should handle duplicates by only counting the latest one received per attester.

I believe this is where the LMD (latest message driven) aspect of the fork choice shines through. We just feed it all valid messages, and it handles new/duplicate/updated attestations.

Reply:
I see, thanks for the clarification